### PR TITLE
SOLR-12708: Aggregate failures from downstream async jobs; add error …

### PR DESCRIPTION
**Pull request by manokovacs**
…handling for RestoreCmd

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/cloud/api/collections/RestoreCmd.java [hunk 1]
```diff
@@ -271,7 +278,37 @@ public void call(ClusterState state, ZkNodeProps message, NamedList results) thr
           propMap.put(ASYNC, asyncId);
         }
         ocmh.addPropertyParams(message, propMap);
-        ocmh.addReplica(clusterState, new ZkNodeProps(propMap), new NamedList(), null);
+        final NamedList addResult = new NamedList();
+        ocmh.addReplica(clusterState, new ZkNodeProps(propMap), addResult, () -> {
+          countDownLatch.countDown();
```
Comment (tflobbe) [inline solr/core/src/java/org/apache/solr/cloud/api/collections/RestoreCmd.java]: I think we should move this to the end of the callback, or there can be a race condition

File: solr/core/src/java/org/apache/solr/cloud/api/collections/RestoreCmd.java [hunk 2]
```diff
@@ -271,7 +278,37 @@ public void call(ClusterState state, ZkNodeProps message, NamedList results) thr
...
+              results.add("success", success);
+            }
+            success.addAll((NamedList) addResult.get("success"));
+          }
+        });
+      }
+
+      boolean allIsDone = countDownLatch.await(1, TimeUnit.HOURS);
+      if (!allIsDone) {
+        throw new TimeoutException("Initial replicas were not created within 10 minutes. Timing out.");
```
Comment (tflobbe) [inline solr/core/src/java/org/apache/solr/cloud/api/collections/RestoreCmd.java]: Outdated error, this is now 1 hour

File: solr/core/src/test-files/solr/configsets/cloud-minimal-faulty/conf/schema.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (tflobbe) [inline solr/core/src/test-files/solr/configsets/cloud-minimal-faulty/conf/schema.xml]: Maybe there is a way to not have to create this new configset? either use the Config API, or the configset API, or maybe just updating one file? or maybe set a system property that one of the existing configsets will use to something invalid?

**Comment by janhoy** [comment]
Closing since JIRA is closed

**Pull request closed without merging**